src/components/TodoList/TodoList.tsx

```diff
@@ -1,44 +1,172 @@
-              button
-            "
-            type="button"
-          >
-            User&nbsp;#1
-          </button>
-        </li>
+interface Props {
+  selectedUserId: number;
+  onUserSelect: (userId: number) => void;
+}
+
+export class TodoList extends React.Component<Props, State> {
+  state: State = {
+    todos: [],
+    searchQuery: '',
+    filterByStatus: '' as FilterByStatusTypes,
+    hasLoadingError: false,
+  };
+
+  async componentDidMount() {
+    try {
+      const todos = await getTodos();
+
+      this.setState({ todos });
+    } catch {
+      this.setState({ hasLoadingError: true });
+    }
+  }
+
+  handleSearch = ({ target }: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({ searchQuery: target.value });
+  };
+
+  handleSelect = ({ target }: React.ChangeEvent<HTMLSelectElement>) => {
+    this.setState({ filterByStatus: target.value as FilterByStatusTypes });
+  };
+
+  randomize = () => {
+    this.setState(prevState => ({
+      todos: [...prevState.todos].sort(() => Math.random() - 0.5),
+    }));
+  };
+
+  getFilteredTodos = () => {
+    const { searchQuery, todos, filterByStatus } = this.state;
+    const queryLow = searchQuery.toLowerCase();
 
-        <li className="TodoList__item TodoList__item--checked">
-          <label>
-            <input type="checkbox" checked readOnly />
-            <p>distinctio vitae autem nihil ut molestias quo</p>
-          </label>
+    const filteredTodos = todos.filter(todo => {
+      switch (filterByStatus) {
+        case FilterByStatusTypes.active:
+          return !todo.completed;
+
+        case FilterByStatusTypes.completed:
+          return todo.completed;
+
+        case FilterByStatusTypes.all:
+        default:
+          return todo;
+      }
+    });
+
+    return filteredTodos.filter(todo => (
+      todo.title
+      && todo.title.toLowerCase().includes(queryLow)
+    ));
+  };
+
+  render() {
+    const {
+      searchQuery,
+      filterByStatus,
+      hasLoadingError,
+    } = this.state;
+    const { selectedUserId, onUserSelect } = this.props;
+    const filteredTodos = this.getFilteredTodos();
+
+    return (
+      <div className="TodoList">
+        <h2>Todos:</h2>
+
+        <div className="TodoList__controls">
+          <input
+            value={searchQuery}
+            type="text"
+            placeholder="Search by title"
+            onChange={this.handleSearch}
+          />
 
           <button
-            className="TodoList__user-button button"
             type="button"
+            className="button"
+            onClick={this.randomize}
           >
-            User&nbsp;#2
+            Randomise
           </button>
-        </li>
-      </ul>
-    </div>
-  </div>
-);
+
+          <select
+            onChange={this.handleSelect}
+            value={filterByStatus}
+          >
+            <option value="" disabled>
+              Choose status type
+            </option>
```

Comment:
it's no sense to have this option, because when you will choose All - nothing gonna be changed